### PR TITLE
feat: create setup resolver UI

### PR DESCRIPTION
**Pull request by luizmuraro**
closes OP-000

## Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Performance improvement
- [ ] Code refactoring

## Description 

- Create edit resolver UI
- Add input component
- Add card component

## Test Instructions (if applicable)

1. Run the project
2. Access `/setup-resolver`

## Screenshots or videos (if applicable)

| Setup resolver | 
| ------------- | 
|      image         |

### CODE REVIEW

File: frontend/src/app/setup-resolver/(components)/SetupResolver.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+              <label
+                htmlFor="resolver"
+                className="text-sm font-medium text-gray-700"
+              >
+                Namespace Offchain Resolver:
+              </label>
+              <Input
+                id="resolver"
+                value={resolver}
+                onChange={(e) => setResolver(e.target.value)}
```
Comment (ribeirojose) [inline frontend/src/app/setup-resolver/(components)/SetupResolver.tsx:42]: to we need to manage this state if this is part of a form?

File: frontend/src/app/setup-resolver/page.tsx [hunk 2]
```diff
@@ -1,7 +1,36 @@
-export default function Page() {
+"use client";
+import React from "react";
+import SetupResolver from "./(components)/SetupResolver";
+
+const Page = () => {
+  const [resolver, setResolver] = React.useState(
+    "0x1234567891234567891234567891234567891234567",
```
Comment (ribeirojose) [inline frontend/src/app/setup-resolver/page.tsx:7]: 👀 I assume that's going to be replaced, but just checking

File: frontend/src/app/setup-resolver/page.tsx [hunk 3]
```diff
@@ -1,7 +1,36 @@
...
+    // Basic validation for hex address format
+    if (!/^0x[0-9a-fA-F]+$/.test(resolver)) {
+      setError(
+        "Invalid resolver address format. Must start with 0x and contain only hexadecimal characters.",
+      );
+      return;
+    }
+
+    setError("");
+    console.log("Submitted resolver:", resolver);
```
Comment (ribeirojose) [inline frontend/src/app/setup-resolver/page.tsx:23]: 👁️

File: frontend/src/app/setup-resolver/types.ts [hunk 4]
```diff
@@ -0,0 +1,6 @@
+export interface SetupResolverProps {
+  resolver: string;
+  setResolver: (value: string) => void;
+  error: string;
+  handleSubmit: (event: React.FormEvent<HTMLFormElement>) => void;
+}
```
Comment (ribeirojose) [inline frontend/src/app/setup-resolver/types.ts:6]: why set this props type in a different file from the component?

**Review by ribeirojose — APPROVED** [review]
lgtm, left a few minor notes 👍

File: frontend/src/app/setup-resolver/(components)/SetupResolver.tsx (hunk 1 above)
Comment (devjoaov) [inline frontend/src/app/setup-resolver/(components)/SetupResolver.tsx:42]: it should be a readonly input right

File: frontend/src/app/setup-resolver/page.tsx [hunk 5]
```diff
@@ -1,7 +1,36 @@
...
+  );
+  const [error, setError] = React.useState("");
+
+  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    // Basic validation for hex address format
+    if (!/^0x[0-9a-fA-F]+$/.test(resolver)) {
+      setError(
+        "Invalid resolver address format. Must start with 0x and contain only hexadecimal characters.",
```
Comment (devjoaov) [inline frontend/src/app/setup-resolver/page.tsx:17]: the user will only setup the namespace resolver, so we don't need this here

**Review by devjoaov — APPROVED** [review]
Looking good. Move the input to be readonly

File: frontend/src/app/setup-resolver/(components)/SetupResolver.tsx (hunk 1 above)
Comment (luizmuraro) [reply frontend/src/app/setup-resolver/(components)/SetupResolver.tsx:42]: Removed on next PR

File: frontend/src/app/setup-resolver/types.ts (hunk 4 above)
Comment (luizmuraro) [reply frontend/src/app/setup-resolver/types.ts:6]: Nothing especial, just to keep the component file cleaner, it looks better for me.

**Pull request merged**